zedc/src/status.rs

```diff
@@ -0,0 +1,304 @@
+        match pm::handle_cmd(vec!["--version".to_owned()]) {
+            Ok(output) => {
+                self.pm_version = Some(output);
+            }
+            Err(e) => {
+                eprintln!("Failed to check package manager version: {}", e);
+            }
+        }
+
+        Ok(())
+    }
+
+    fn check_zowe_cli(&mut self) -> Result<()> {
+        match cmd::as_binary("zowe")
+            .arg("--version")
+            .current_dir(&self.workspace_path)
+            .output()
+        {
+            Ok(output) => {
+                if output.status.success() {
+                    self.zowe_cli_version =
+                        Some(String::from_utf8_lossy(&output.stdout).trim().to_string());
+                }
+            }
+            Err(e) => {
+                eprintln!("Failed to check Zowe CLI version: {}", e);
+            }
+        }
+        Ok(())
+    }
+
+    fn check_vscode(&mut self) -> Result<()> {
+        match cmd::as_binary("code").arg("--version").output() {
+            Ok(output) => {
+                if output.status.success() {
+                    self.vscode_version =
+                        Some(String::from_utf8_lossy(&output.stdout).trim().to_string());
+                }
+            }
+            Err(e) => {
+                eprintln!("Failed to check VS Code version: {}", e);
+            }
+        }
+        Ok(())
+    }
+
+    fn check_git_status(&mut self) -> Result<()> {
+        match Command::new("git")
+            .arg("status")
+            .arg("--porcelain")
+            .output()
+        {
+            Ok(output) => {
+                if output.status.success() {
+                    let status = String::from_utf8_lossy(&output.stdout);
+                    if !status.trim().is_empty() {
+                        self.git_status = Some(status.trim().to_string());
+                    }
+                }
+            }
+            Err(e) => {
+                eprintln!("Failed to check Git status: {}", e);
+            }
+        }
+        Ok(())
+    }
+
+    fn check_env_vars(&mut self) {
+        // List of relevant environment variables to check
+        let relevant_vars = [
+            "ZEDC_PAT",
+            "NODE_ENV",
+            "HOME",
+            "VSCODE_EXTENSIONS",
+            "ZOWE_CLI_HOME",
+            "ZOWE_OPT_USER",
+            "ZOWE_OPT_PASS",
+        ];
+
+        for var in relevant_vars {
+            if let Ok(value) = std::env::var(var) {
+                // Mask sensitive values
+                let display_value = if var.contains("PASS") || var.contains("PAT") {
+                    "********".to_string()
+                } else {
+                    value
+                };
+                self.env_vars.push((var.to_string(), display_value));
+            }
+        }
+    }
+
+    fn check_zowe_env(&mut self) -> Result<()> {
+        match cmd::as_binary("zowe")
+            .args(["config", "report-env"])
```

Comment:
I see that we are running a config report-env command.
I like that the `zedc status` command supports the `-v | --verbose` flag, and it prints the output of the `zowe config report-env` command 😋 

A few things we could improve upon is displaying any`ZOWE_` env vars 😋 
Also any installed VSCEs (ideally filtered by the ones that have the ze-api as a dependency) 😋